### DISCUSSION TITLE
[RFC] Zero-config

### DISCUSSION THREAD

**Discussion by jaredpalmer** [vercel/turborepo#7253] [Ideas]
### Describe the feature you'd like to request

Go back to zero config

### Describe the solution you'd like

- [ ] Remove the need for `turbo.json` or any config and assume every task in topological by default (like lerna)
- [ ] Make caching opt-in (remove the defaults) or detected
	- [ ] 	Write a codemod to take existing undefined `outputs` in each `pipeline` key and set them as `['dist/**', 'build/**']` (which is the current default)

### Describe alternatives you've considered

N/A

**Comment by samchouse** [comment]
Will `turbo.json` still be used for advanced configs such as those that have tasks depend on others, change base branch and global dependencies?

**Comment by jlarmstrongiv** [comment]
I do actually like that turbo fails when a command is not in the `turbo.json`.  It ensures I set the dependencies and cache right and rename my scripts correctly.
